pkg/keymanagementprovider/azurekeyvault/provider.go

```diff
@@ -140,25 +165,48 @@ func (s *akvKMProvider) GetCertificates(ctx context.Context) (map[keymanagementp
 	for _, keyVaultCert := range s.certificates {
 		logger.GetLogger(ctx, logOpt).Debugf("fetching secret from key vault, certName %v,  keyvault %v", keyVaultCert.Name, s.vaultURI)
 
-		// fetch the object from Key Vault
-		// GetSecret is required so we can fetch the entire cert chain. See issue https://github.com/ratify-project/ratify/issues/695 for details
 		startTime := time.Now()
+
+		// GetSecret is required so we can fetch the entire cert chain. See issue https://github.com/ratify-project/ratify/issues/695 for details
 		secretBundle, err := s.kvClient.GetSecret(ctx, s.vaultURI, keyVaultCert.Name, keyVaultCert.Version)
 		if err != nil {
+			// certificate is disabled, remove it from the map
+			var de autorest.DetailedError
+			if errors.As(err, &de) {
+				var re *azure.RequestError
+				if errors.As(de.Original, &re) {
+					if re.ServiceError.Code == "SecretDisabled" {
+						certBundle, err := s.kvClient.GetCertificate(ctx, s.vaultURI, keyVaultCert.Name, keyVaultCert.Version)
+						if err != nil {
+							return nil, nil, fmt.Errorf("failed to get certificate objectName:%s, objectVersion:%s, error: %w", keyVaultCert.Name, keyVaultCert.Version, err)
+						}
+						keyVaultCert.Version = getObjectVersion(*certBundle.Kid)
+						isEnabled := *certBundle.Attributes.Enabled
```

Comment:
curious if `isEnabled` would be always `false` in this case?

Reply:
Correct, it will always be false for this code path.